### PR TITLE
fix targetOriginAllowList parsing

### PR DESCRIPTION
**Pull request by alexander-chervony**

### CODE REVIEW

File: gateway.go [hunk 1]
```diff
@@ -100,6 +100,9 @@ func (s *gatewayResource) gatewayHandler(w http.ResponseWriter, r *http.Request)
 		if err == ConfigMismatchError {
 			http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
 			return
+		if err == TargetForbiddenError {
```
Comment (izhamoidsin) [inline gateway.go:103]: Looks like you forgot to close `}` after the previous `if` section

**Review by izhamoidsin — CHANGES_REQUESTED** [review]

**Review by izhamoidsin — APPROVED** [review]

File: gateway.go [hunk 2]
```diff
@@ -100,6 +100,9 @@ func (s *gatewayResource) gatewayHandler(w http.ResponseWriter, r *http.Request)
 		if err == ConfigMismatchError {
 			http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
 			return
+		} else if err == TargetForbiddenError {
```
Comment (chris-wood) [inline gateway.go:103]: This change shouldn't be necessary since`TargetForbiddenError` will always be encapsulated.

File: handler.go [hunk 3]
```diff
@@ -253,6 +253,7 @@ func (h FilteredHttpRequestHandler) Handle(req *http.Request, metrics Metrics) (
 		_, ok := h.allowedOrigins[req.Host]
 		if !ok {
 			metrics.Fire(metricsResultTargetRequestForbidden)
+			log.Printf("TargetForbiddenError: %s", req.Host)
```
Comment (chris-wood) [inline handler.go:256]: Can we remove this log line?

**Review by chris-wood — CHANGES_REQUESTED** [review]
Nice catch! I have two nits that I think we need to address before merging.

File: handler.go (hunk 3 above)
Comment (alexander-chervony) [reply handler.go:256]: We use it to be able to identify and fix not-allowed requests for the content. It happens that some services use direct links to our S3 web wrapper service. And we wanto to log these cases and fix as time permits.

Comment (chris-wood) [reply handler.go:256]: Yeah, that's a fine use case, but it should be done in a separate PR. I also wonder if there's a better way to track these beyond logging (and then scraping the logs).

Comment (alexander-chervony) [reply handler.go:256]: but we could put it under if (VERBOSE). Will it work for you?

File: gateway.go (hunk 2 above)
Comment (chris-wood) [reply gateway.go:103]: I don't think this is true. This error is always encapsulated and only known to the client.

File: handler.go (hunk 3 above)
Comment (chris-wood) [reply handler.go:256]: That might work. Can we do this in a separate PR?

**Review by chris-wood — APPROVED** [review]

**Pull request merged**